### PR TITLE
[Feature](agg-state) OLAP_FIELD_TYPE_AGG_STATE support more serialized_type

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
OLAP_FIELD_TYPE_AGG_STATE support more serialized_type

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: regression-test/suites/mv_p0/agg_state/diffrent_serialize/diffrent_serialize.groovy [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    sql "insert into d_table select 1,1,1,'a';"
+    sql "insert into d_table select 1,2,1,'a';"
+
+    explain {
+        sql("select k1,bitmap_count(bitmap_agg(k2)) from d_table group by k1 order by 1;")
+        contains "(mv1)"
+    }
+    qt_select_mv "select k1,bitmap_count(bitmap_agg(k2)) from d_table group by k1 order by 1;"
+
+/*
```
Comment (HappenLee) [inline regression-test/suites/mv_p0/agg_state/diffrent_serialize/diffrent_serialize.groovy:68]: why comment the case?

File: be/src/vec/olap/olap_data_convertor.cpp [hunk 2]
```diff
@@ -1045,15 +1054,17 @@ Status OlapBlockDataConvertor::OlapColumnDataConvertorMap::convert_to_olap(
...
     _key_convertor->set_source_column(key_typed_column, start_offset, elem_size);
     RETURN_IF_ERROR(_key_convertor->convert_to_olap());
 
-    ColumnWithTypeAndName value_typed_column = {value_data, data_type_map->get_value_type(),
+    ColumnWithTypeAndName value_typed_column = {value_data, _data_type.get_value_type(),
                                                 "map.value"};
     _value_convertor->set_source_column(value_typed_column, start_offset, elem_size);
     RETURN_IF_ERROR(_value_convertor->convert_to_olap());
 
+    LOG(WARNING) << _data_type.get_name();
```
Comment (HappenLee) [inline be/src/vec/olap/olap_data_convertor.cpp:1066]: seems dubeg logging, delete it

File: be/src/vec/olap/olap_data_convertor.cpp [hunk 3]
```diff
@@ -181,23 +202,17 @@ OlapBlockDataConvertor::create_olap_column_data_convertor(const TabletColumn& co
...
     }
     case FieldType::OLAP_FIELD_TYPE_MAP: {
-        const auto& key_column = column.get_sub_column(0);
-        const auto& value_column = column.get_sub_column(1);
-        return std::make_unique<OlapColumnDataConvertorMap>(
-                create_olap_column_data_convertor(key_column),
-                create_olap_column_data_convertor(value_column));
+        return create_map_convertor(column);
     }
     default: {
```
Comment (HappenLee) [inline be/src/vec/olap/olap_data_convertor.cpp:210]: no need the the `default` if do nothing

File: be/src/vec/olap/olap_data_convertor.cpp [hunk 4]
```diff
@@ -74,6 +73,39 @@ void OlapBlockDataConvertor::add_column_data_convertor(const TabletColumn& colum
...
+OlapBlockDataConvertor::create_agg_state_convertor(const TabletColumn& column) {
+    auto data_type = DataTypeFactory::instance().create_data_type(column);
+    const auto* agg_state_type = assert_cast<const vectorized::DataTypeAggState*>(data_type.get());
+    auto type = agg_state_type->get_serialized_type()->get_type_as_type_descriptor().type;
+
+    if (type == PrimitiveType::TYPE_STRING) {
+        return std::make_unique<OlapColumnDataConvertorVarChar>(false);
+    } else if (type == PrimitiveType::TYPE_OBJECT) {
+        return std::make_unique<OlapColumnDataConvertorBitMap>();
+    } else if (type == PrimitiveType::INVALID_TYPE) {
```
Comment (HappenLee) [inline be/src/vec/olap/olap_data_convertor.cpp:100]: """
The logic appears rather strange and requires a significant amount of commentary.
"""

File: be/src/vec/data_types/data_type_factory.cpp [hunk 5]
```diff
@@ -604,7 +619,7 @@ DataTypePtr DataTypeFactory::create_data_type(const segment_v2::ColumnMetaPB& pc
         names.reserve(col_size);
         for (size_t i = 0; i < col_size; i++) {
             dataTypes.push_back(create_data_type(pcolumn.children_columns(i)));
-            names.push_back("");
+            names.emplace_back("");
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type_factory.cpp:622]: just resize and not need call `emplace_back`

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 6]
```diff
@@ -72,13 +74,135 @@
...
+
+    std::unique_ptr<ColumnReader> null_reader;
+    if (meta.is_nullable()) {
+        RETURN_IF_ERROR(ColumnReader::create(opts, meta.children_columns(2),
+                                             meta.children_columns(2).num_rows(), file_reader,
+                                             &null_reader));
+    }
+
+    // The num rows of the array reader equals to the num rows of the length reader.
+    num_rows = meta.children_columns(1).num_rows();
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:109]: the `num_rows` is param, but set value in the function ?

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 7]
```diff
@@ -708,6 +719,79 @@ Status ColumnReader::new_iterator(ColumnIterator** iterator) {
...
+    if (!_agg_state_ptr) { // meet old version ColumnMetaPB
+        *iterator = new FileColumnIterator(this);
+        return Status::OK();
+    }
+
+    const auto* agg_state_type =
+            assert_cast<const vectorized::DataTypeAggState*>(_agg_state_ptr.get());
+    auto type = agg_state_type->get_serialized_type()->get_type_as_type_descriptor().type;
+
+    if (type == PrimitiveType::TYPE_STRING || type == PrimitiveType::INVALID_TYPE ||
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:732]:
```type == PrimitiveType::TYPE_STRING || type == PrimitiveType::INVALID_TYPE ||
        type == PrimitiveType::TYPE_OBJECT``` better be a function

File: regression-test/suites/mv_p0/agg_state/diffrent_serialize/diffrent_serialize.groovy (hunk 1 above)
Comment (BiteTheDDDDt) [reply regression-test/suites/mv_p0/agg_state/diffrent_serialize/diffrent_serialize.groovy:68]: This part will be implemented in the next PR

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/column_writer.cpp [hunk 8]
```diff
@@ -88,267 +88,305 @@ class NullBitmapBuilder {
...
+    for (uint32_t i = 0; i < column->get_subtype_count(); i++) {
+        const TabletColumn& sub_column = column->get_sub_column(i);
+
+        // create sub writer
+        ColumnWriterOptions column_options;
+        column_options.meta = opts.meta->mutable_children_columns(i);
+        column_options.need_zone_map = false;
+        column_options.need_bloom_filter = sub_column.is_bf_column();
+        column_options.need_bitmap_index = sub_column.has_bitmap_index();
+        if (sub_column.type() == FieldType::OLAP_FIELD_TYPE_STRUCT) {
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/column_writer.cpp:107]: should be a function, repeated too many time

File: be/src/olap/rowset/segment_v2/column_writer.cpp [hunk 9]
```diff
@@ -88,267 +88,305 @@ class NullBitmapBuilder {
...
+        std::unique_ptr<ColumnWriter> sub_column_writer;
+        RETURN_IF_ERROR(
+                ColumnWriter::create(column_options, &sub_column, file_writer, &sub_column_writer));
+        sub_column_writers.push_back(std::move(sub_column_writer));
+    }
+
+    // if nullable, create null writer
+    ScalarColumnWriter* null_writer = nullptr;
+    if (opts.meta->is_nullable()) {
+        FieldType null_type = FieldType::OLAP_FIELD_TYPE_TINYINT;
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/column_writer.cpp:132]: should we simple the to a function, also do the work in array writer

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**